man/systemd-escape.xml

```diff
@@ -159,6 +174,14 @@ tmp-waldi-foobar.mount</programlisting>
     <para>To generate instance names of three strings:</para>
     <programlisting>$ systemd-escape --template=systemd-nspawn@.service 'My Container 1' 'containerb' 'container/III'
 systemd-nspawn@My\x20Container\x201.service systemd-nspawn@containerb.service systemd-nspawn@container-III.service</programlisting>
+
+    <para>To extract the instance part of an instantiated unit:</para>
+    <programlisting>$ systemd-escape -u --instance 'systemd-nspawn@My\x20Container\x201.service'
+My Container 1</programlisting>
+
+    <para>To extract the instance part of an instance of a particular template:</para>
+    <programlisting>$ systemd-escape -u --template=systemd-nspawn@.service 'systemd-nspawn@My\x20Container\x201.service'
+My Container 1</programlisting>
```

Comment:
My mental model is: `--unescape` exactly reverses the operation of `systemd-escape`. Passing a valid unit through `systemd-escape -u` and then `systemd-escape` with the same options should result in no change from the original input, and passing a string through `systemd-escape` and then `systemd-escape -u` with the same options should also result in no change – even if those options included `--template`!

`--path`, for example, will usually remove a leading slash. With `--unescape`, it will instead add it. `--template`, then, will usually add the template to to the instance. With `--unescape`, it will remove it again.

And it’s not a complete duplicate of `--instance`: it also asserts that the template is the one you expect. I think this is a very useful check – if your script takes a machine instance and also does some `machinectl` operations on the machine name (extracted with `systemd-escape`), you wouldn’t want to do those operations if you got a non-`systemd-nspawn@.service` unit for whatever reason. (I’m ignoring non-`systemd-nspawn` containers and VMs for this example.)

But with your suggestion of `--unescape`-less `--instance`, which does sound useful as well, that nice and clean model kinda breaks apart. I don’t want to just remove `--template` support from the PR, but I’m not sure how all the bits should go together now…